app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/step/choose/ChooseInsuranceToTerminateViewModel.kt

```diff
@@ -0,0 +1,183 @@
+    var currentSelectedId by remember { mutableStateOf(insuranceId) }
+    var terminationStep: TerminateInsuranceStep? by remember { mutableStateOf(null) }
+    var currentPartialState by remember {
+      mutableStateOf(lastState.toPartialState())
+    }
+
+    CollectEvents { event ->
+      when (event) {
+        ChooseInsuranceToTerminateEvent.RetryLoadData -> {
+          loadIteration++
+        }
+
+        is ChooseInsuranceToTerminateEvent.SelectInsurance -> {
+          currentSelectedId = event.insurance.id
+          if (currentPartialState is PartialUiState.Success) {
+            currentPartialState =
+              (currentPartialState as PartialUiState.Success).copy(selectedInsurance = event.insurance)
+          }
+        }
+      }
+    }
+
+    LaunchedEffect(currentSelectedId) {
+      val id = currentSelectedId
+      if (id != null) {
+        terminationStep = terminateInsuranceRepository.startTerminationFlow(InsuranceId(id)).getOrNull()
+        logcat { "mariia: LaunchedEffect(currentSelectedId) finished, terminationStep: $terminationStep" }
+      }
+    }
+
+    LaunchedEffect(loadIteration) {
+      currentPartialState = PartialUiState.Loading
+      combine(
+        flow { emit(getContractsToTerminateUseCase.invoke()) },
+        featureManager.isFeatureEnabled(Feature.TERMINATION_FLOW),
+      ) { contractsResult, isTerminationFlowEnabled ->
+        contractsResult to isTerminationFlowEnabled
+      }.collect { (contractsResult, isTerminationFlowEnabled) ->
+        logcat { "mariia: contractsResult: $contractsResult isTerminationFlowEnabled: $isTerminationFlowEnabled" }
+
+        if (!isTerminationFlowEnabled) {
+          currentPartialState = PartialUiState.NotAllowed
+        } else {
+          contractsResult.fold(
+            ifLeft = {
+              logcat(priority = LogPriority.INFO) { "Cannot load contracts for cancellation" }
+              logcat { "mariia: LaunchedEffect(loadIteration) Cannot load contracts for cancellation" }
+              currentPartialState = PartialUiState.Failure
+            },
+            ifRight = { eligibleInsurances ->
+              logcat { "mariia: LaunchedEffect(loadIteration) Successfully loaded contracts for cancellation" }
+              logcat(priority = LogPriority.INFO) { "Successfully loaded contracts for cancellation" }
+              val selectedInsurance = if (insuranceId != null) {
+                eligibleInsurances.firstOrNull { it.id == insuranceId }
+              } else {
+                null
+              }
+              currentPartialState = if (eligibleInsurances.isNotEmpty()) {
+                PartialUiState.Success(
+                  eligibleInsurances,
+                  selectedInsurance,
+                )
+              } else {
+                PartialUiState.NotAllowed
+                // todo: if we somehow get an empty list, I think we should show the same
+                // "you don't have eligible contracts blablabla" copy
```

Comment:
```suggestion
                PartialUiState.NotAllowed
```
Yep I agree, this feels like it can be the same fallback of telling them that they got nothing to cancel and to show the copy we have + a button to the chat perhaps.

So with that said since it's more of a scenario where they are not eligible, and not that they are not allowed, could we name the state as `PartialUiState.NotEligible`?